mirrord/agent/src/main.rs

```diff
@@ -71,93 +74,78 @@ mod watched_task;
 const CHANNEL_SIZE: usize = 1024;
 
 /// Keeps track of connected clients.
-/// If pausing target, also pauses and unpauses when number of clients changes from or to 0.
 #[derive(Debug)]
 struct State {
     clients: HashSet<ClientId>,
     index_allocator: IndexAllocator<ClientId>,
-    /// Was the pause argument passed? If true, will pause the container when no clients are
-    /// connected.
-    should_pause: bool,
-    /// This is an option because it is acceptable not to pass a container runtime and id if not
-    /// pausing. When those args are not passed, container is None.
-    container: Option<Container>,
-}
-
-/// This is to make sure we don't leave the target container paused if the agent hits an error and
-/// exits early without removing all of its clients.
-impl Drop for State {
-    fn drop(&mut self) {
-        if self.should_pause && !self.no_clients_left() {
-            info!(
-                "Agent exiting without having removed all the clients. Unpausing target container."
-            );
-            if let Err(err) = executor::block_on(self.container.as_ref().unwrap().unpause()) {
-                error!(
-                    "Could not unpause target container while exiting early, got error: {err:?}"
-                );
-            }
-        }
-    }
+    /// Handle to the target container if there is one.
+    container: Option<ContainerHandle>,
+    env: HashMap<String, String>,
+    /// Whether this agent is run in an ephemeral container.
+    ephemeral: bool,
 }
 
 impl State {
-    /// Returns Err if container runtime operations failed or if the `pause` arg was passed, but
-    /// the container info (runtime and id) was not.
+    /// Return [`Err`] if container runtime operations failed.
     pub async fn new(args: &Args) -> Result<State> {
         let container =
             get_container(args.container_id.as_ref(), args.container_runtime.as_ref()).await?;
 
-        if container.is_none() && args.pause {
-            Err(AgentError::MissingContainerInfo)?
-        }
+        let handle = match container {
```

Comment:
As far as I can see, we don't use the original `container` after this, so why not:

```suggestion
        let container = match container {
```